azurerm/resource_arm_data_lake_store.go

```diff
@@ -0,0 +1,193 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/datalake/store/mgmt/2016-11-01/account"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/response"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDataLakeStore() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDateLakeStoreCreate,
+		Read:   resourceArmDateLakeStoreRead,
+		Update: resourceArmDateLakeStoreUpdate,
+		Delete: resourceArmDateLakeStoreDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"tier": {
+				Type:             schema.TypeString,
+				ForceNew:         false,
+				Optional:         true,
+				Default:          string(account.Consumption),
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(account.Consumption),
+					string(account.Commitment1TB),
+					string(account.Commitment10TB),
+					string(account.Commitment100TB),
+					string(account.Commitment500TB),
+					string(account.Commitment1PB),
+					string(account.Commitment5PB),
+				}, true),
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceArmDateLakeStoreCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).dataLakeStoreAccountClient
+	ctx := meta.(*ArmClient).StopContext
+
+	log.Printf("[INFO] preparing arguments for Azure ARM Date Lake Store creation.")
+
+	name := d.Get("name").(string)
+	location := azureRMNormalizeLocation(d.Get("location").(string))
+	resGroup := d.Get("resource_group_name").(string)
+	tier := d.Get("tier").(string)
+	tags := d.Get("tags").(map[string]interface{})
+
+	dateLakeStore := account.CreateDataLakeStoreAccountParameters{
+		Location: &location,
+		Tags:     expandTags(tags),
+		CreateDataLakeStoreAccountProperties: &account.CreateDataLakeStoreAccountProperties{
+			NewTier: account.TierType(tier),
+		},
+	}
+
+	future, err := client.Create(ctx, resGroup, name, dateLakeStore)
+	if err != nil {
+		return err
```

Comment:
Could we wrap these errors? like how you are in the update
```
future, err := client.Update(ctx, resourceGroup, name, props)
	if err != nil {
		return fmt.Errorf("Error issuing update request for Data Lake Store %q (Resource Group %q): %+v", name, resourceGroup, err)
	}
```

Reply:
done for Create